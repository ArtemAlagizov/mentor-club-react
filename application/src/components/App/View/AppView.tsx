import React from "react";
import { BrowserRouter as Router, Route, Redirect } from "react-router-dom";
import LoginPage from "../../LoginPage/Container/LoginPage";
import PrivateRoute from "../../PrivateRoute/Container/PrivateRoute";
import MentorRequestListPage from "../../MentorRequestListPage/Container/MentorRequestListPage";
import AuthenticatedPage from "../../AuthenticatedPage/Container/AuthenticatedPage";
import { IAppViewProps } from "./IAppViewProps";
import MentorRequestPage from "../../MentorRequestCreationPage/Container/MentorRequestCreationPage";
import MentorListPage from "../../MentorListPage/Container/MentorListPage";

const AppView: React.FC<IAppViewProps> = ({ logout }) => (
  <Router basename="/mentor-club">
    <PrivateRoute>
      <AuthenticatedPage logout={logout}>
        <Route exact path="/">
          <Redirect to="/mentor-list" />
        </Route>
        <Route exact path="/mentor-requests">
          <MentorRequestListPage />
        </Route>
        <Route exact path="/mentor-list">
          <MentorListPage />
        </Route>
        <Route exact path="/mentor-requests/new">
          <MentorRequestPage />
        </Route>
      </AuthenticatedPage>
    </PrivateRoute>
    <Route exact path="/login">
      <LoginPage />
    </Route>
  </Router>
);

export default AppView;
