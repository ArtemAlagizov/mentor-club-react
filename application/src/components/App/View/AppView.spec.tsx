import { shallow } from "enzyme";
import React from "react";
import AppView from "./AppView";

describe("Given AppView", () => {
  describe("when rendering", () => {
    it("should have AppView matching snapshot", () => {
      const appView = shallow(<AppView logout={jest.fn()} />);

      expect(appView).toMatchSnapshot();
    });
  });
});
