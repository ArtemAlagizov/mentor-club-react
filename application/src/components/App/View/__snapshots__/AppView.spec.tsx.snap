// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given AppView when rendering should have AppView matching snapshot 1`] = `
<BrowserRouter
  basename="/mentor-club"
>
  <Connect(PrivateRoute)>
    <Connect(AuthenticatedPage)
      logout={[MockFunction]}
    >
      <Route
        exact={true}
        path="/"
      >
        <Redirect
          to="/mentor-requests"
        />
      </Route>
      <Route
        exact={true}
        path="/mentor-requests"
      >
        <Connect(MentorRequestListPage) />
      </Route>
      <Route
        exact={true}
        path="/mentor-requests/new"
      />
    </Connect(AuthenticatedPage)>
  </Connect(PrivateRoute)>
  <Route
    exact={true}
    path="/login"
  >
    <Connect(LoginPage) />
  </Route>
</BrowserRouter>
`;
